internal/app/secretless/handlers/mysql/handler.go

```diff
@@ -89,37 +168,69 @@ func stream(source, dest net.Conn, callback func([]byte)) {
 	}
 }
 
-// Pipe performs continuous bidirectional transfer of data between the client and backend.
-func (h *Handler) Pipe() {
+// pipe performs continuous bidirectional transfer of data between the client and backend.
+//
+func (h *Handler) pipe() {
 	plugin_v1.PipeHandlerWithStream(h, stream, h.EventNotifier, func() {
 		h.ShutdownNotifier(h)
 	})
 }
 
-// Run configures the backend connection info, connects to the backend to
-// complete the connection phase, and pipes the data between the client and
-// the backend
+// This is the main handler method. It:
+//
+// 1. Fetches credentials from the provider.
+// 2. Dials the backend.
+// 3. Runs through the connection phase steps to authenticate.
+// 4. Pipes all future bytes unaltered between client and server.
+//
 func (h *Handler) Run() {
-	var err error
 
-	if err = h.ConfigureBackend(); err != nil {
-		h.abort(err)
+	// Upgrade to a decorated connection that handles protocol details for us
+	// We need to do this first because sendErrorToClient uses this to send error messages.
+	//
+	h.mySQLClientConn = NewMySQLClientConnection(h.ClientConnection)
+
+	// 1. Fetches credentials from the provider.
+	//
+	connDetails, err := h.fetchConnectionDetails()
+	if err != nil {
+		h.sendErrorToClient(err)
+		return
+	}
+
+	// 2. Dials the backend.
+	//
+	rawBackendConn, err := net.Dial("tcp", connDetails.Address())
+	if err != nil {
+		h.sendErrorToClient(err)
 		return
 	}
 
-	if err = h.ConnectToBackend(); err != nil {
-		h.abort(err)
+	h.mySQLBackendConn = NewMySQLBackendConnection(rawBackendConn)
+
+	// 3. Runs through the connection phase steps to authenticate.
+	//
+	connPhase := NewConnectionPhase(h.mySQLClientConn, h.mySQLBackendConn, connDetails)
```

Comment:
this naming seems off to me. what it's doing is negotiating the authentication handshake to create a new authenticated connection. wdyt about `ConnectionHandshake` instead of `ConnectionPhase`?

Reply:
or maybe `AuthenticationHandshake`